index.js

```diff
@@ -323,41 +435,169 @@ Arduino.prototype.reboot = function(cb) {
   }
 };
 
+Arduino.prototype._computeChecksum = function (message) {
+  var self = this;
+  // Compute the checksum by XORing all the character values in the string.
+  var checksum = 0;
+
+  for (var i = 0; i < message.length; i++) {
+    checksum = checksum ^ (self.options.binary ? message[i] : message.charCodeAt(i));
+  }
+
+  if (!self.options.binary) {
+    // Convert it to hexadecimal (base-16, upper case, most significant nybble
+    // first).
+    checksum = Number(checksum).toString(16).toUpperCase();
+    if (checksum.length < 2) {
+      checksum = ("00" + checksum).slice(-2);
+    }
+  }
+
+
+  return checksum;
+};
+
+/**
+ * Write to the arduino and waits for the acknowledgment bit to be returned. Currently only has support
+ * for binary sending.
+ * @param message {Buffer} - Binary buffer with message to send
+ * @param cb {Function} - Callback
+ * @param options {Object} - Options for sending
+ * @param options.attempts {Number} - Number of times to re-attempt sending. Default is 10.
+ *                                    0 means won't send.
+ * @param options.priority {String} - Priority: "high" or "low"
+ * @param options.timeout {Number} - Amount of time to wait for acknowledgment before resending. Default is 200 ms
+ */
+Arduino.prototype.writeAndWaitForAcknowledgement = function (message, cb = () => {}, options) {
```

Comment:
Default parameters are a very newly supported es6 feature while the rest of this library is es5.